contracts/exchange/TokenPriceProvider.sol

```diff
@@ -14,19 +15,17 @@ contract TokenPriceProvider {
 
     mapping(address => uint256) public cachedPrices;
 
-    function syncPrice(ERC20 token) public {
-        uint256 expectedRate;
-        (expectedRate,) = kyberNetwork().getExpectedRate(token, ERC20(ETH_TOKEN_ADDRESS), 10000);
-        cachedPrices[address(token)] = expectedRate;
+    function setPrice(ERC20 _token, uint256 _price) external onlyManager {
+        cachedPrices[address(_token)] = _price;
     }
 
-    //
-    // Convenience functions
-    //
+    function syncPrice(ERC20 _token) external onlyManager {
```

Comment:
Why restrict this function to managers? It's ok if external parties synch the price of Kyber tokens so we should remove the `onlyManager` modifier.

Reply:
What would be the benefits of doing that? I feel that if we open price changes to any address we increase the attack surface of this contract for no good reason. We would also need to make sure in `_syncPrice()` to reject changes of prices that try to set a price to zero (otherwise prices of tokens not on Kyber can be set to 0 using `syncPrice`). That would prevent us from handling tokens whose value is zero (probably never going to happen anyway but who knows). This also makes the contract less predictable: we set a price to 10 and shortly after someone execute `syncPrice` to set it to 10.9. I had left `syncPrice` here just to facilitate the transition from the previous TokenPriceProvider but I'm now thinking that it would make more sense to remove all Kyber-syncing methods.

Reply:
At first I did not see the point of keeping the possibility to synch with Kyber, but then I realized it could be a way to make the system more trustless for our users, i.e. if a user does not trust the prices that we push or if we are corrupted he can make the oracle synch with Kyber.

But I am equally fine with removing it. So I think we should either remove this method or make it public.